scheduler/src/cook/mesos/util.clj

```diff
@@ -395,22 +399,25 @@
    and timeframe, without a custom executor.
    Note that this query is not performant for completed jobs, use
    get-completed-jobs-by-user instead."
-  [db user start end state name-filter-fn include-custom-executor?]
+  [db user start end state name-filter-fn include-custom-executor? pool-name]
   (let [state-keyword (case state
                         "running" :job.state/running
                         "waiting" :job.state/waiting)
+        default-pool? (pool/default-pool? pool-name)
+        pool-name' (or pool-name pool/nil-pool)
         jobs
         (timers/time!
           (timers/timer ["cook-mesos" "scheduler" (str "get-" (name state) "-jobs-by-user-duration")])
           (->> (q '[:find [?j ...]
-                    :in $ ?user ?state ?start ?end
+                    :in $ ?user ?state ?start ?end ?pool-name ?default-pool
                     :where
                     [?j :job/state ?state]
                     [?j :job/user ?user]
                     [?j :job/submit-time ?t]
                     [(<= ?start ?t)]
-                    [(< ?t ?end)]]
-                  db user state-keyword start end)
+                    [(< ?t ?end)]
+                    [(cook.mesos.pool/check-pool-for-listing $ ?j :job/pool ?pool-name ?default-pool)]]
```

Comment:
Probably doesn't matter now, but should we think about putting this first instead of last and indexing on it to help smaller pools?

Reply:
Putting the pools first would probably help in clusters with many pools of differing sizes, but it would probably hurt clusters with no pools and many jobs in one status.